### PR TITLE
DM-30284: Accommodations towards Gen2/Gen3 parity and determinism in coaddition

### PR DESCRIPTION
**Pull request by laurenam**

### CODE REVIEW

File: python/lsst/coadd/utils/getExposureId.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        if patchIndexStr == patch:
+            patchNumId = tractInfo.getSequentialPatchIndex(patchInfo)
+            break
+    try:
+        expId, maxBits = skyMap.pack_data_id(tract, patchNumId, band=band)
+    except Exception as e:
+        if log is not None:
+            log.warn("Setting exposureId to match Gen3 failed with: %s.  Falling back to "
+                     "Gen2 implementation.", e)
+            expId = int(dataRef.get(coaddName + "CoaddId"))
```
Comment (TallJimbo) [inline python/lsst/coadd/utils/getExposureId.py]: I think you need to either de-indent this line (so it fires if `log is None`) or add a bare `raise` after the `if` block; as is, if `log` is `None`, you'll get `NameError("name 'expId' is not defined")` when the function tries to return.

But if - as I suspect - the real case is that `log` is never `None`, maybe just remove that default from the argument, and remove the `if` block, too?

File: python/lsst/coadd/utils/getExposureId.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["getExposureId"]
+
+from lsst.daf.persistence import NoResults
+
+
+def getExposureId(dataRef, coaddName="deep", includeBand=True, log=None):
```
Comment (TallJimbo) [inline python/lsst/coadd/utils/getExposureId.py]: Feel free to ignore this as this function won't last long, but `getGen3CoaddExposureId` or similar would be a lot more descriptive.

**Review by TallJimbo — APPROVED** [review]

File: python/lsst/coadd/utils/getExposureId.py (hunk 1 above)
Comment (laurenam) [reply python/lsst/coadd/utils/getExposureId.py]: Thanks for the catch!  I went with the de-indent as I could imagine a context for not having a `log` (and think I had one a some point!)

File: python/lsst/coadd/utils/getExposureId.py (hunk 2 above)
Comment (laurenam) [reply python/lsst/coadd/utils/getExposureId.py]: Renamed as suggested.

**Pull request merged**